services/app-api/postgres/submissionWithRevisionsHelpers.ts

```diff
@@ -47,12 +47,24 @@ function convertToSubmission2Type(dbSub: StateSubmissionWithRevisions): Submissi
             let submitInfo: UpdateInfoType | undefined = undefined
             if (r.submittedAt) {
                 submitInfo = {
-                    updatedAt: r.submittedAt
+                    updatedAt: r.submittedAt,
```

Comment:
Can we do the same handling for `submitInfo` as we are doing for `unlockInfo` (make sure we use the previous values if they exist(.  I know thats not entirely  in this ticket but with this change we would start rewriting submitInfo with an empty string for certain values whenever this function is called.

Reply:
I wanted to get some clarification. Did you mean to add two fields like `submittedBy` and `submittedChange` in postgress that correspond with `updatedBy` and `updatedReason` ? Also I'm confused about what the previous values are.

Reply:
I meant do something like this for submitInfo
```
submitInfo = {
                    updatedAt: r.submittedt,
                    updatedBy: r.submittedBy,
                    updatedReason: r.submittedReason,
                }
                }
```

instead of writing empty strings there. This isn't blocking though I could be misunderstanding something. It just caught my eye the use of empty strings here rewriting over other data.

Reply:
Ah ok I understand.

I believe in order for me to do that I would need to add `submittedBy` and `submittedReason` to postgres since the `StateSubmissionRevision` (auto generated from prisma.schema?) does not include those two.  Does that sound correct @macrael 

I initially set empty strings there because `UpdateInfoType` in `Submission2Type.d.ts` required all three properties. I have no problems with adding those two fields into postgres and updating everything else to accommodate those changes.